packages/ketcher-core/src/application/editor/modes/SequenceMode.ts

```diff
@@ -480,54 +480,65 @@ export class SequenceMode extends BaseMode {
   }
 
   private get keyboardEventHandlers() {
-    return {
-      delete: {
-        shortcut: ['Backspace', 'Delete'],
-        handler: () => {
-          const editor = CoreEditor.provideEditorInstance();
-          const previousNode = SequenceRenderer.previousNode;
-          const previousCaretPosition = SequenceRenderer.caretPosition;
-          const selections = SequenceRenderer.selections;
-          const modelChanges = new Command();
-          let nodesToDelete: NodesSelection;
-
-          if (selections.length) {
-            modelChanges.merge(this.deleteSelectedDrawingEntities());
-            nodesToDelete = selections;
-          } else if (previousNode) {
-            previousNode.monomers.forEach((monomer) => {
-              modelChanges.merge(
-                editor.drawingEntitiesManager.deleteMonomer(monomer),
-              );
-            });
-            nodesToDelete = [
-              [
-                {
-                  node: previousNode,
-                  nodeIndexOverall:
-                    SequenceRenderer.previousCaretPosition as number,
-                },
-              ],
-            ];
-          } else {
-            return;
-          }
+    const deleteNode = (direction: 'left' | 'right') => {
+      const editor = CoreEditor.provideEditorInstance();
+      const nodeToDelete =
+        direction === 'left'
+          ? SequenceRenderer.previousNode
+          : SequenceRenderer.getNodeByPointer(SequenceRenderer.caretPosition);
+      const caretPosition =
+        direction === 'left'
+          ? (SequenceRenderer.previousCaretPosition as number)
+          : SequenceRenderer.caretPosition;
+      const selections = SequenceRenderer.selections;
+      const modelChanges = new Command();
+      let nodesToDelete: NodesSelection;
+
+      if (selections.length) {
+        modelChanges.merge(this.deleteSelectedDrawingEntities());
+        nodesToDelete = selections;
+      } else if (nodeToDelete) {
+        nodeToDelete.monomers.forEach((monomer) => {
+          modelChanges.merge(
+            editor.drawingEntitiesManager.deleteMonomer(monomer),
+          );
+        });
+        nodesToDelete = [
+          [
+            {
+              node: nodeToDelete,
+              nodeIndexOverall: caretPosition,
+            },
+          ],
+        ];
+      } else {
+        return;
+      }
 
-          modelChanges.merge(this.handleNodesDeletion(nodesToDelete));
+      modelChanges.merge(this.handleNodesDeletion(nodesToDelete));
 
-          this.finishNodesDeletion(
-            modelChanges,
-            previousCaretPosition,
-            nodesToDelete[0][0].nodeIndexOverall,
-          );
+      this.finishNodesDeletion(
+        modelChanges,
+        nodesToDelete[0][nodesToDelete[0].length - 1].nodeIndexOverall + 1,
+        nodesToDelete[0][0].nodeIndexOverall,
+      );
 
```

Comment:
Looks like this change causes bug found by autotest. I believe we need to fix it here instead of update screenshot.